app/src/models/areaTemplateRelation.model.ts

```diff
@@ -0,0 +1,14 @@
+import mongoose from "mongoose";
+
+const { Schema } = mongoose;
+
+const AreaTemplateRelationSchema = new Schema({
+  areaId: { type: String },
+  templateId: { type: String }
+});
+
+AreaTemplateRelationSchema.index({ areaId: 1 });
+
+export const AreaTemplateRelationModel = mongoose.model("AreaTemplateRelation", AreaTemplateRelationSchema);
```

Comment:
Check this out when you need typescript support for the model!

[https://github.com/Automattic/mongoose/issues/5046#issuecomment-412114160](https://github.com/Automattic/mongoose/issues/5046#issuecomment-412114160)

`"@types/mongoose": "4.7.15",`